ping/ping.go

```diff
@@ -0,0 +1,212 @@
+package ping
+
+import (
+	"fmt"
+	"math/rand"
+	"sort"
+	"time"
+
+	"github.com/go-ping/ping"
+
+	"github.com/vertcoin-project/one-click-miner-vnext/logging"
+	"github.com/vertcoin-project/one-click-miner-vnext/networks"
+	"github.com/vertcoin-project/one-click-miner-vnext/util"
+)
+
+//To add your node to the list simply add it to the nodes variable with the correct syntax and change the index number to relect the current number of nodes
+//Your node can have a combined max fee of currently 2%, it will not be selected if it doesn't meet that requirement
+//Your node needs to be pingable (Allow ICMP pings)
+//Your node should be live 24/7 to avoid disrupting miners.
+
+var nodes = []string{
+	"fr1.vtconline.org",
+	"p2proxy.vertcoin.org",
+	"mindcraftblocks.com",
+	"vtc-fl.javerity.com",
+	"vtc-ca.javerity.com",
+	"vtc.p2pminers.nl",
+	"p2p-usa.xyz",
+	"p2p-ekb.xyz",
+	"asia.p2p-spb.xyz",
+	"p2p-spb.xyz",
+	"p2p-south.xyz",
+	"siberia.mine.nu",
+}
+var NodeResults = make(map[time.Duration]string)
```

Comment:
You could make a slice of those structs, fill the most recent ping time, and then call `sort.Slice` to sort it by ping time.

Reply:
This makes sense to do.

> In any case using duration as map key seems like a potential collision if two servers have the same ping time.

This would be extremely rare though given the precision of the assigned time.Duration variable.